Tests/Pcap++Test/Tests/FilterTests.cpp

```diff
@@ -656,14 +656,42 @@ PTF_TEST_CASE(TestPcapFiltersOffline)
 	}
 	rawPacketVec.clear();
 
+	//------------------------------
+	// Composite filter - Proto + IP
+	//------------------------------
+	pcpp::IPFilter ipFilter("10.0.0.6", pcpp::SRC);
+	protoFilter.setProto(pcpp::UDP);
+	std::vector<pcpp::GeneralFilter*> filterVec;
+	filterVec.push_back(&ipFilter);
+	filterVec.push_back(&protoFilter);
+	pcpp::CompositeFilter compositeFilter(filterVec, " and ");
+	compositeFilter.parseToString(filterAsString);
+	PTF_ASSERT_EQUAL(filterAsString, "(ip and src net 10.0.0.6) and (udp)")
+
+	compositeFilter.removeFilter(&ipFilter);
+	compositeFilter.parseToString(filterAsString);
+	PTF_ASSERT_EQUAL(filterAsString, "(udp)");
+
+	compositeFilter.addFilter(&ipFilter);
+	compositeFilter.parseToString(filterAsString);
+	PTF_ASSERT_EQUAL(filterAsString, "(udp) and (ip and src net 10.0.0.6)")
+
+	compositeFilter.clearAllFilters();
+	compositeFilter.parseToString(filterAsString);
+	PTF_ASSERT_EQUAL(filterAsString, "")
+
+	compositeFilter.setFilters(filterVec);
+	compositeFilter.parseToString(filterAsString);
+	PTF_ASSERT_EQUAL(filterAsString, "(ip and src net 10.0.0.6) and (udp)")
 
 	//-----------------------
 	//And filter - Proto + IP
 	//-----------------------
 
-	pcpp::IPFilter ipFilter("10.0.0.6", pcpp::SRC);
+	ipFilter.setAddr("10.0.0.6");
```

Comment:
let's not reuse the variable, which makes it hard to understand the intention.
please separate the variable for " Composite filter - Proto + IP" and "And filter - Proto + IP"

Reply:
Reused them because `ipFilter` and `filterVec` seemed to be reused from the And Filter test into the Complex Filter test so I figured that was fine.

While on the subject, if variable reuse was discouraged across tests, would it not be better to have each test in a block scope, that way they can't be reused accidentally?

Reply:
Resolved in 34a9f04